android/src/main/java/com/mattermost/networkclient/Helper.kt

```diff
@@ -17,10 +26,65 @@ fun Response.returnAsWriteableMap (): WritableMap {
     return map;
 }
 
-fun Response.handleViaPromise(promise: Promise){
-    return if(this.isSuccessful){
+/**
+ * Completes the response handling by calling a promise resolve / reject based on response type
+ *
+ * @param promise The promise to resolve/reject
+ */
+fun Response.promiseResolution(promise: Promise): Response {
+    if (this.isSuccessful) {
         promise.resolve(this.returnAsWriteableMap());
     } else {
         promise.reject(this.code.toString(), this.returnAsWriteableMap())
     }
+    return this
+}
+
+/**
+ * Parses options passed in over the bridge for individual requests
+ *
+ * @param options ReadableMap of options from the App
+ */
+fun Request.Builder.parseOptions(options: ReadableMap): Request.Builder {
+    return this
+}
+
+/**
+ * Parses options passed in over the bridge for client sessions
+ *
+ * @params options ReadableMap of options from the App
+ */
+fun OkHttpClient.Builder.parseOptions(options: ReadableMap): OkHttpClient.Builder {
+    this.followRedirects(true);
+    this.followSslRedirects(true);
+
+    val followRedirects = options.getBoolean("followRedirects")
+    if (followRedirects) {
```

Comment:
would this evaluate if the value is `false` ? seems that is only being  set when the option is set to `true`

Reply:
Oof good spot; this is turning into a catch-22; on the android side `options.getBoolean()` is a non-null, so if it doesn't exist it will return false ergo that if statement doesn't get called. Originally I had it as `if(followRedirects != null)` and that was always true...

@migbot I think we'll have to manually set `followRedirects = true` on the JS side as a "default" when initialising a client to get around this.

Reply:
I found a `options.hasKey()` which is the same idea; using that instead (thanks!).